plugin/trino-kafka-event-listener/src/main/java/io/trino/plugin/eventlistener/kafka/KafkaEventListener.java

```diff
@@ -49,20 +49,11 @@ public KafkaEventListener(KafkaEventListenerConfig config, KafkaProducerFactory
         publishSplitCompletedEvent = config.getPublishSplitCompletedEvent();
         isAnonymizationEnabled = config.isAnonymizationEnabled();
 
-        try {
-            if (publishCreatedEvent || publishCompletedEvent) {
-                kafkaPublisher = new KafkaEventPublisher(config, producerFactory, stats);
-            }
-            else {
-                LOG.warn("Event listener will be no-op, as neither created events nor completed events are published.");
-            }
+        if (publishCreatedEvent || publishCompletedEvent) {
+            kafkaPublisher = new KafkaEventPublisher(config, producerFactory, stats);
         }
-        catch (Exception e) {
-            if (config.getTerminateOnInitializationFailure()) {
```

Comment:
see https://github.com/trinodb/trino/pull/23606#issuecomment-2383686412. I don't really think we need to complicate it further